Content.Shared/Humanoid/Prototypes/RandomHumanoidPrototype.cs

```diff
@@ -0,0 +1,30 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Array;
+
+namespace Content.Shared.Humanoid.Prototypes;
+
+[Prototype("randomHumanoid")]
+public sealed class RandomHumanoidPrototype : IPrototype, IInheritingPrototype
+{
+    [IdDataField] public string ID { get; } = default!;
+
+    [ParentDataField(typeof(PrototypeIdArraySerializer<RandomHumanoidPrototype>))]
+    public string[]? Parents { get; }
+
+    [AbstractDataField]
+    public bool Abstract { get; }
+
+    [DataField("randomizeName")] public bool RandomizeName { get; } = true;
+
+    /// <summary>
+    ///     Species that will be ignored by the randomizer.
+    /// </summary>
+    [DataField("speciesBlacklist")]
+    public HashSet<string> SpeciesBlacklist { get; } = new();
+
+    /// <summary>
+    ///     Extra components to add to this entity.
+    /// </summary>
+    [DataField("components")]
+    public EntityPrototype.ComponentRegistry? Components { get; }
```

Comment:
Getting the ComponentRegistry in scope is a bit strange, you can look at how it's done in TraitPrototype.cs, where a strange looking ```using``` was used, although I'm not sure that's a better way. Some maintainer should probably figure out what's going on with that.

Reply:
see https://github.com/space-wizards/RobustToolbox/issues/3284